storage/s3_test.go

```diff
@@ -398,3 +405,203 @@ func TestS3Retry(t *testing.T) {
 		})
 	}
 }
+
+// Credit to aws-sdk-go
+func val(i interface{}, s string) interface{} {
+	v, err := awsutil.ValuesAtPath(i, s)
+	if err != nil || len(v) == 0 {
+		return nil
+	}
+	if _, ok := v[0].(io.Reader); ok {
+		return v[0]
+	}
+
+	if rv := reflect.ValueOf(v[0]); rv.Kind() == reflect.Ptr {
+		return rv.Elem().Interface()
+	}
+
+	return v[0]
+}
+
+func setExpectedErrs(t *testing.T, expected string, got interface{}) {
+	if got == nil {
+		if expected != "" {
+			t.Errorf("Expected %q, but received %q", "", got)
+		}
+	} else if expected != got {
```

Comment:
I think `else` could be removed, we could just check if they are equal or not.

Reply:
in this part, since we are comparing interface{} with string, we would like to consider empty string and nil interface as being equal

Reply:
What I meant was something like;

 ```go
func setExpectedErrs(t *testing.T, expected string, got interface{}) {
	if got == nil {
		if expected != "" {
			t.Errorf("Expected %q, but received %q", "", got)
			return
		}
	}

	if expected != got {
		t.Errorf("Expected %q, but received %q", expected, got)
		return
	}
}
```

Reply:
oh ok, thanks!